pkg/apb/client.go

```diff
@@ -227,3 +228,21 @@ func OcLogin(log *logging.Logger, args ...string) error {
 	log.Debug(string(output))
 	return nil
 }
+
+// TODO: Instead of putting namespace directly as a parameter, we should create a dictionary
+// of apb_metadata and put context and other variables in it so we don't pollute the user
+// parameter space.
+func createExtraVars(context *Context, parameters *Parameters) (string, error) {
+	var paramsCopy Parameters
+	if parameters != nil && *parameters != nil {
```

Comment:
He has to check both, because a pointer to a map, is not very useful. that really should be ```parameters Parameters``` and not a pointer as maps are [already what you would be looking for](https://dave.cheney.net/2017/04/30/if-a-map-isnt-a-reference-variable-what-is-it) with a pointer.